pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -1309,3 +1332,79 @@ func initializeAnnotations(cd *hivev1.ClusterDeployment) {
 		cd.Annotations = map[string]string{}
 	}
 }
+
+// managePullSecrets() takes pull secretes in to account and returns the merged pull request
+func (r *ReconcileClusterDeployment) managePullSecrets(cd *hivev1.ClusterDeployment, cdLog log.FieldLogger) (string, error) {
+
+	// For code readability let's call the pull secret in cluster deployment config as local pull secret
+	localPullSecret, err := controllerutils.LoadSecretData(r.Client, cd.Spec.PullSecret.Name, cd.Namespace, corev1.DockerConfigJsonKey)
+	if err != nil {
+		cdLog.Warn("unable to load pull secret from secret")
+	}
+
+	// Check if global pull secret from env as it comes from hive config
+	globalPullSecret := os.Getenv("GLOBAL_PULL_SECRET")
+	if len(globalPullSecret) == 0 {
+		cdLog.Warn("global pull secret not available")
+		if localPullSecret == "" {
+			errMsg := "Pull request or global pull request not available"
+			cdLog.Error(errMsg)
+			return "", fmt.Errorf("%s", errMsg)
+		}
+	}
+
+	// If global pull secret is present then merge the pullSecret and globalPullSecret
+	if globalPullSecret != "" && localPullSecret != "" {
```

Comment:
You can combine this block and the previous block into one switch statement.

```
switch {
case globalPullSecret != "" && localPullSecret != "":
  // Merge
case globalPullSecret != "":
  // Use global
case localPulSecret != "":
  // Use local
default:
  // Error since neither are present
}
```